ActionPlans/Start-DlToO365GroupUpgradeChecks.ps1

```diff
@@ -0,0 +1,427 @@
+$ConditionMemberRestriction|Add-Member -NotePropertyName "Member Join Restriction" -NotePropertyValue $dg.MemberJoinRestriction
+$ConditionMemberRestriction|Add-Member -NotePropertyName "Member Depart Restriction" -NotePropertyValue $dg.MemberDepartRestriction
+[string]$SectionTitle = "Validating Distribution Group Member Restriction Properties"
+[string]$Description = "Checking if Distribution Group can't be upgraded if MemberJoinRestriction or MemberDepartRestriction or both values are set to Closed"        
+if ($dg.MemberJoinRestriction -eq "Open" -and $dg.MemberDepartRestriction -eq "Open") {
+    [PSCustomObject]$ConditionMemberRestrictionHTML = Prepare-ObjectForHTMLReport -SectionTitle $SectionTitle -SectionTitleColor "Green" -Description $Description -DataType "ArrayList" -EffectiveDataArrayList $ConditionMemberRestriction -TableType "Table"
+    $null = $TheObjectToConvertToHTML.Add($ConditionMemberRestrictionHTML)  
+    } 
+    else {
+    $DGConditionsmet++    
+    [PSCustomObject]$ConditionMemberRestrictionHTML = Prepare-ObjectForHTMLReport -SectionTitle $SectionTitle -SectionTitleColor "Red" -Description $Description -DataType "ArrayList" -EffectiveDataArrayList $ConditionMemberRestriction -TableType "Table"
+    $null = $TheObjectToConvertToHTML.Add($ConditionMemberRestrictionHTML)
+    }
+#endRegion Check if Distribution Group can't be upgraded because Member*Restriction is set to "Closed"
+
+#Region Check if Distribution Group can't be upgraded because it is DirSynced
+$ConditionIsDirSynced=New-Object PSObject    
+$ConditionIsDirSynced|Add-Member -NotePropertyName "IsDirSynced" -NotePropertyValue $dg.IsDirSynced
+[string]$SectionTitle = "Validating Distribution Group IsDirSynced Property"
+[string]$Description = "Checking if Distribution Group can't be upgraded because IsDirSynced value is true"    
+if ($dg.IsDirSynced -eq $true) {
+    $DGConditionsmet++ 
+    [PSCustomObject]$ConditionIsDirSyncedHTML = Prepare-ObjectForHTMLReport -SectionTitle $SectionTitle -SectionTitleColor "Red" -Description $Description -DataType "ArrayList" -EffectiveDataArrayList $ConditionIsDirSynced -TableType "List"
+    $null = $TheObjectToConvertToHTML.Add($ConditionIsDirSyncedHTML)
+
+} 
+else {
+    [PSCustomObject]$ConditionIsDirSyncedHTML = Prepare-ObjectForHTMLReport -SectionTitle $SectionTitle -SectionTitleColor "Green" -Description $Description -DataType "ArrayList" -EffectiveDataArrayList $ConditionIsDirSynced -TableType "Table"
+    $null = $TheObjectToConvertToHTML.Add($ConditionIsDirSyncedHTML)
+}
+#endRegion Check if Distribution Group can't be upgraded because it is DirSynced
+
+#region Check if Distribution Group can't be upgraded because EmailAddressPolicyViolated
+$eap = Get-EmailAddressPolicy -ErrorAction stop
+[string]$SectionTitle = "Validating Distribution Group matching EmailAddressPolicy"
+[string]$Description = "Checking if Distribution Group can't be upgraded because of matching EmailAddressPolicy"
```

Comment:
can you improve a little bit the description to be more clear?

Reply:
I've edited the description to
"Checking if Distribution Group can't be upgraded because Admin has applied Group Email Address Policy for the groups on the organization"